pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -409,22 +409,31 @@ type ServiceInstance struct {
 	Status ServiceInstanceStatus `json:"status"`
 }
 
-// ServiceInstanceSpec represents the desired state of an Instance.
-type ServiceInstanceSpec struct {
+// DesiredPlan defines the user specification for the desired
+// ServicePlan and ServiceClass. Because there are multiple ways to
+// specify the desired Class/Plan, this structure specifies the
```

Comment:
can you elaborate on what are the multiple ways are and how the wrapper helps?

Reply:
Oops, I missed this, my apologies. Yes, so see #1285 for example. So, we either add 2 more fields to ServiceInstanceSpec or refactor at this point. Earlier there have also been discussions about using things like selectors, adding more and more fields to this Spec seems like messy way to go.